### ISSUE TITLE
Slowdowns or crashes when parsing email with attachments

### ISSUE THREAD

**Issue by azulli** [Enough-Software/enough_mail#85]
Hi,
using this library in a mobile app, I have slowdowns or even crashes fetching medium to large emails with attachments via IMAP.
For emails about 200k in size the delay is somewhat noticeable, but eventually completes successfully.
Increasing the attachment size, it needs even minutes between receiving data and returning a MimeMessage instance.
For attachments over 5-6Mb often the VM is killed by the os process manager, even when running in the emulator.

The huge cpu usage and slowdown exist even when parsing a sample email in pure dart on pc.

**Comment by robert-virkus** [comment]
Thanks for the report, it would be great if you could record problematic `FETCH` calls and send them to me, so that I can evaluate them better. In general the API is not yet performance optimized at all, so any help here is very much appreciated.

**Comment by azulli** [comment]
Unfortunately I'm unable to provide an imap log, instead i've crafted a eml message that can be used to highlight the MimeMessage build issue (the other part is the proper `FETCH` parsing) . I'ts almost 8mb so I'm unsure how to provide it to you, either attaching it here or using a file hoster.
I've also done a major rework that resulted in a sensible speed boost and no apparent crashes until today, published in to my "performances" branch (it's a work in progress).
Long story short, I've converted all bodyRaw in Uint8List, using sublistView to reference the relative parts of each parent object, reducing the memory usage of the string copy operations, and applied some fixes.

**Comment by robert-virkus** [comment]
Hi @azulli please open pull requests if you have some optimizations, thanks! The latest 0.1.0 release also features some performance and memory improvements, along with all the API changes.
